consensus/replay.go

```diff
@@ -257,8 +257,10 @@ func (h *Handshaker) Handshake(proxyApp proxy.AppConns) error {
 	)
 
 	// Set AppVersion on the state.
-	h.initialState.Version.Consensus.App = version.Protocol(res.AppVersion)
-	sm.SaveState(h.stateDB, h.initialState)
+	if h.initialState.Version.Consensus.App != version.Protocol(res.AppVersion) {
```

Comment:
Do we even need the if statement here?

Reply:
This judge can avoid we save state every time when do the handshake, if the app version not change, I think it's no need we save state here